scripts/cloud-test.py

```diff
@@ -0,0 +1,213 @@
+# -----------------------------------------------------------------------------------------------------------
+# Copyright (c) 2020 PyInstaller Development Team.
+#
+# This file is distributed under the terms of the GNU General Public License (version 2.0 or later).
+#
+# The full license is available in LICENSE.GPL.txt, distributed with this software.
+#
+# SPDX-License-Identifier: GPL-2.0-or-later
+# -----------------------------------------------------------------------------------------------------------
+"""
+A command line interface to launch the 'Oneshot test' workflow (.github/workflows/oneshot-test.yml).
+"""
+
+import os
+from pathlib import Path
+import subprocess
+import re
+from pprint import pformat
+import textwrap
+
+try:
+    import click
+    import github
+
+except ImportError:
+    raise SystemExit("The script requires PyGithub and click. Please run the following then re-run this script\n\n"
+                     "pip install click PyGithub\n")
+
+
+HERE = Path(__file__).parent
+
+
+def authenticated_user():
+    """Get a logged in Github user."""
+    # Anyone know a better way of doing this?
+    token = os.environ.get("GITHUB_TOKEN")
+
+    if token is None:
+        token = input("Please enter a Github authentication token or your username and password separated by a space.\n"
+                      "Alternatively you may set the GITHUB_TOKEN environment variable instead:\n")
+
+    user = github.Github(*token.split(maxsplit=1))
+    try:
+        user.get_user().login
+    except github.BadCredentialsException:
+        raise SystemExit("Authentication failed due to invalid credentials.")
+
+    print("Authenticated successfully.")
+    return user
+
+
+def _get_current_branch():
+    """Get the branch currently active locally in git.
+
+    The location of this repo is defined based on the location of this script so it is current-working-dir
+    independent.
+    """
+    branch_command = "git -C \"{}\" branch".format(HERE)
+    branch_output = subprocess.check_output(branch_command).decode()

```

Comment:
This does not work on linux - it raises `FileNotFoundError`.

Either `shell=True` needs to be added, or `branch_command` needs to be in the argument-list form (which is probably the preferable approach).

Reply:
OK. I had a weird issue in another project where a mac CI stubbornly refused to accept the argument-list form which is why I've been wary of it here. I'll stick with adding `shell=True` unless you're sure mac will be ok with the argument-list form?

Reply:
On macOS `FileNotFoundError` is raised as well, but the argument-list form seems to work for me.

Reply:
It also saves you the trouble of quoting the path (and, if I recall correctly, different escaping rules between platforms/shell interpreters).